File: docs/test-cases/exceptional-conditions.py
# Test case: exceptional-conditions (A10:2025)
import traceback
from flask import Flask, jsonify

app = Flask(__name__)

@app.route("/process")
def process():
    try:
        data = fetch_data()
        result = transform(data)
        return jsonify(result)
    except Exception as e:
        # BUG: exposes full stack trace to client
        return jsonify({"error": traceback.format_exc()}), 500

@app.route("/admin/delete")
def delete_resource():
    try:
        resource_id = get_resource_id()
        db.delete(resource_id)
    except Exception:
        # BUG: fail-open — returns success even on error
        pass
    return jsonify({"status": "ok"})

@app.errorhandler(404)
def not_found(e):
    # BUG: exposes internal framework version info
    return jsonify({"error": str(e), "server": "Flask/2.3", "python": "3.11"}), 404

File: docs/test-cases/exceptional-conditions.java
// Test case: exceptional-conditions (A10:2025)
package com.example.api;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
public class AdminController {

    private final AuthService auth = new AuthService();
    private final AdminService admin = new AdminService();

    public boolean isAuthorized(String token, String resource) {
        try {
            return auth.checkPermission(token, resource);
        } catch (Exception e) {
            // BUG: fail-open — any exception grants access
            return true;
        }
    }

    @GetMapping("/admin/report")
    public ResponseEntity<String> report(@RequestHeader("X-Token") String token) {
        try {
            String data = admin.generateReport(token);
            return ResponseEntity.ok(data);
        } catch (Exception e) {
            // BUG: stack trace / internal message leaked to client
            return ResponseEntity.status(500).body(e.toString());
        }
    }

    @PostMapping("/admin/purge")
    public ResponseEntity<String> purge(@RequestHeader("X-Token") String token) {
        try {
            admin.purgeAll(token);
        } catch (Throwable t) {
            // BUG: swallows Throwable including OutOfMemoryError and StackOverflowError
        }
        return ResponseEntity.ok("done");
    }
}
